### PR TITLE
Ext_proc unit test fuzzer leaks buffer memory

### PR DESCRIPTION
**Pull request by yanjunxiang-google**
Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by yanjunxiang-google** [comment]
In ext_proc fuzzing code, the chunk->data below is never drained, which caused memory leaking.

https://github.com/envoyproxy/envoy/blob/9a1e2948e48c5e53079a48cccc94f1ef04a51a6d/source/extensions/filters/http/ext_proc/processor_state.cc#L307

It is not a problem in real case since the future filter chain which call decodeData() will drain the data.

**Review by KBaichoo — APPROVED** [review]

File: test/extensions/filters/http/ext_proc/unit_test_fuzz/ext_proc_unit_test_fuzz.cc [hunk 1]
```diff
@@ -30,6 +30,12 @@ class FuzzerMocks {
     ON_CALL(encoder_callbacks_, encodingBuffer()).WillByDefault(Return(&buffer_));
     ON_CALL(decoder_callbacks_, decoderBufferLimit()).WillByDefault(Return(1024));
     ON_CALL(encoder_callbacks_, encoderBufferLimit()).WillByDefault(Return(1024));
+    ON_CALL(decoder_callbacks_, injectDecodedDataToFilterChain(_, _))
```
Comment (tyxia) [inline test/extensions/filters/http/ext_proc/unit_test_fuzz/ext_proc_unit_test_fuzz.cc:33]: qq: Is this something can be done in the centralized place /test/mocks/http/mocks.h? Other filter can run into same issue if have similar behavior?

**Review by tyxia — COMMENTED** [review]
Thanks!

File: test/extensions/filters/http/ext_proc/unit_test_fuzz/ext_proc_unit_test_fuzz.cc (hunk 1 above)
Comment (yanjunxiang-google) [reply test/extensions/filters/http/ext_proc/unit_test_fuzz/ext_proc_unit_test_fuzz.cc:33]: Very few filters call this inject data function.  So, keep it within individual filter which uses it make more sense to me.

**Pull request merged**